metashare/api/views.py

```diff
@@ -171,7 +171,7 @@ class ScratchOrgViewSet(viewsets.ModelViewSet):
     filterset_class = ScratchOrgFilter
 
     def perform_create(self, *args, **kwargs):
-        if self.request.user.is_devhub_enabled:
+        if self.request.user.is_devhub_enabled or self.request.user.devhub_username:
```

Comment:
Wouldn't it make sense to check for devhub_username first so we can avoid is_devhub_enabled's request to Salesforce when it is set? Actually, probably is_devhub_enabled should short-circuit to return True if devhub_username is set.